layerfetcher/layers.go

```diff
@@ -243,15 +220,15 @@ func (l *Logic) receiveLayerHash(id types.LayerID, p p2ppeers.Peer, peers int, d
 	for hash, peer := range hashes {
 		//build receiver function
 		receiveForPeerFunc := func(data []byte) {
-			l.receiveBlockHashes(id, data, nil)
+			l.receiveBlockHashes(ctx, id, data, nil)
 		}
 		remainingPeers := 0
 		errFunc := func(err error) {
-			l.receiveBlockHashes(id, nil, err)
+			l.receiveBlockHashes(ctx, id, nil, err)
 		}
-		err := l.net.SendRequest(LayerHashDB, hash.Bytes(), peer[remainingPeers], receiveForPeerFunc, errFunc)
+		err := l.net.SendRequest(context.TODO(), LayerHashDB, hash.Bytes(), peer[remainingPeers], receiveForPeerFunc, errFunc)
```

Comment:
Either you consistently miss this spot, or I'm missing something?

```suggestion
		err := l.net.SendRequest(ctx, LayerHashDB, hash.Bytes(), peer[remainingPeers], receiveForPeerFunc, errFunc)
```